lib/src/providers/country_provider.dart

```diff
@@ -19,9 +19,10 @@ class CountryProvider {
     if (countries == null || countries.isEmpty) {
       return jsonList.map((country) => Country.fromJson(country)).toList();
     }
-    List filteredList = jsonList.where((country) {
-      return countries.contains(country[PropertyName]);
-    }).toList();
+    List filteredList = countries
+        .map((countryCode) => jsonList.firstWhere((country) => country[PropertyName] == countryCode,
+            orElse: () => null))
+        .toList();
 
     return filteredList.map((country) => Country.fromJson(country)).toList();
```

Comment:
Maybe let's return `Country.fromJson(country)` inside the first loop instead of in another loop on 27?
```suggestion
final filteredList = countries.map((countryCode) {
  final _country = jsonList.firstWhere(
    (country) => country[PropertyName] == countryCode,
    orElse: () => null,
  );

  if (_country == null) return null;

  return Country.fromJson(_country);
}).where((c) => c != null).toList();
```